src/controller/user.controller.ts

```diff
@@ -0,0 +1,123 @@
+import { Request, Response } from "express";
+import { NOT_FOUND, INTERNAL_SERVER_ERROR, OK, CREATED ,SocialMediaType, BAD_REQUEST } from "../utils/constants.utils";
+import userService from "../services/user.service";
+import { ISocialMediaURLS, IUser } from "../models/interfaces.models";
+import { ICreatorWorkMetadata } from "../models/interfaces.models";
+import { forEachChild } from "typescript";
+
+
+class UserController {
+
+    // get all the users
+    getCreatedUser = async (request: Request, response: Response) => {
+        try {
+            const users = await userService.getUser();
+            if (users) {
+                return response.status(OK).send(users);
+            }
+            else {
+                return response.status(NOT_FOUND).send({
+                    message: `User not found`
+                });
+            }
+        } catch (error: any) {
+            return response.status(NOT_FOUND).send({
+                message: `An Error Ocurred: \n${error.message}`
+            });
+        }
+    };
+
+    getCreatedUserByWalletAddress = async (request: Request, response: Response) => {
+        const { walletAddress } = request.params;
+        try {
+            const users = await userService.getUserByWalletAddress(walletAddress as string);
+            if (users) {
+                return response.status(OK).send(users);
+            }
+            else {
+                return response.status(NOT_FOUND).send({
+                    message: `User not found`
+                });
+            }
+        } catch (error: any) {
+            return response.status(INTERNAL_SERVER_ERROR).send({
+                message: `An Error Ocurred: \n${error.message}`
+            });
+        }
+    };
+
+    addUser = async (request: Request, response: Response) => {
```

Comment:
Please rewrite with the Schema Update in mind

Reply:
update the addUser